docs/mesh/domain.md

```diff
@@ -0,0 +1,51 @@
+# Domain
```

Comment:
We should point to this from somewhere in `docs/README.md` Just a link and a sentence under long feature description